src/python/qeqiskit/optimizer/optimizer_test.py

```diff
@@ -40,3 +43,39 @@ def test_optimizer_succeeds_on_cost_function_without_gradient(self):
             self.assertIn("opt_value", results.keys())
             self.assertIn("opt_params", results.keys())
             self.assertIn("history", results.keys())
+
+    def test_optmizer_records_history_if_keep_value_history_is_added_as_option(self):
```

Comment:
Shouldn't this be part of the `OptimizerTests` suite in `z-quantum-core`?

Also, you have a typo in `optmizer` and in the net functio nin `optimzier` :D

Reply:
If we want to include those in Optimizer tests we would have to restructure them, so that e.g. there are two lists of optimizers (one that keeps track of history and one that does not) because we pass `keep_value_history` during optimizer initialization. Another option would be to move `keep_value_history` to `minimize` method itself, so it can be fine-tuned per every optimization. @mstechly what do you think?

Also, typos fixed in 7b9a26d

Reply:
Can't we use the same list of optimizers just to change the `keep_value_history` property of the object at the beginning of the test?

```
for optimizer in self.optimizers:
    optimizer. keep_value_history = True
    # REST OF THE CODE
```
?

Reply:
That's hacky, but it should work :D I suggest getting this merged here, so that the history recording stuff can work for people that use it already. writing a ticket for improving optimizer tests and implementing it as soon as we have time.  Sounds fair?

Reply:
Sounds fair!